### DISCUSSION TITLE
Create a registration form with createUser and a static token.

### DISCUSSION THREAD

**Discussion by webthingz** [Intevel/nuxt-directus#261] [Q&A]
Hello There,

I'm trying to figure out how to create a registration form. I have created an API user in directus for these requests with a static token. With the directus/sdk I could set a static token .with(staticToken('TOKEN') and use this with the request.

So I tried adding the token in my nuxt.config file.
```
runtimeConfig: {
    public: { directus: { url: "https://mydirectus.url/" } },
    token: "my-token",
  },
```
and use the token in my script setup. 
```
const { createUser } = useDirectusAuth();
const { token } = useDirectusToken();

const onSubmit = async () => {
  try {
    const newUser = await createUser({
      email: email.value,
      password: password.value,
    });
  } catch (e) {}
};
```
First of all, i'm still learning here so correct me if there are better and safer ways to do this :-)
It is very hard to find a good example/explanation how to do this also like best practice should be great to find.

When I tried to register a user I always get a 403 that I don't have permission...
So I tried to check if the script is working and when I first login with a admin user I can register a user just fine. 

How can I create users with a secure and best practice in mind?

Hope somebody here can push me in the right direction :-)

Kind Regards,
Ziggy

**Comment by sandros94** [comment]
Ehy @webthingz, to use a static token you should set it either via ENV variables with `NUXT_PUBLIC_DIRECTUS_STATIC_TOKEN` or in `nuxt.config.ts` like:

```diff
runtimeConfig: {
  public: {
    directus: {
      url: "https://mydirectus.url/",
+     staticToken: "my-token",
    },
  },
- token: "my-token",
},
```

`const { token } = useDirectusToken()` is used to retrieve your current, logged in, access token. Not needed for your use case.
If you want to access your static token in Nuxt's runtime config you have to: `const { staticToken } = useRuntimeConfig().public.directus`, but you don't have to most of the times

**Reply by webthingz** [reply]
Thanks for your answer! That was the way I did it also with the directus/sdk. I will check this out now. But you don't want your token to be used public or is that just the name 'Public' that scares me using it like that? There is not much explaining at Directus for creating users in a secure way.. or I'm just to noobie. Maybe it's better to just send and invite link and set permissions within Directus so that public can create invites..  I found out that you only need to give _create_ permissions on _email, status and roles_.  Anyway thanks I will give an update when I got this working with the staticToken.

**Reply by sandros94** [reply]
> I found out that you only need to give create permissions on email, status and roles

Never set permission to `roles`, otherwise others could edit themselves to admin!!

Yes, the `public` nuxt config exposes everything client side, since if the logic to register a user is done in a nuxt page or component this would require it to have the right privileges in the Directus api. If you need to have a private static token (that is not exposed to the client) then you would need to move that logic to [Nuxt's api routes](https://nuxt.com/docs/guide/directory-structure/server#server-routes) but `nuxt-directus@5.6` doesn't have builtin server utils to handle this server side.

> That was the way I did it also with the directus/sdk

If you are familiar with the `directus/sdk` I would suggest you to take a look at the work I've been doing the last few months at #215 by removing `nuxt-directus` and installing `nuxt-directus-next` (also change it in the `nuxt.config.ts` modules). This is the upcoming `v6` version and wraps the directus/sdk, both client and server side (although, for server api routes only the preconfigured `client`s are available).

Documentation is still minimal, since I was trying to fixing up a few missing features, but you can find [it here](https://next.nuxt-directus.site/docs/getting-started/installation). Or better of, take a look at some examples in the [`dev/playground`](https://github.com/Intevel/nuxt-directus/tree/9535b1ef51aa062dbaff3a348c9985a1ca868be9/playground). As long as you use any client or builtin composable from the `nuxt-directus-next` the right token will be picked from your `.env` or `nuxt.config.ts`
